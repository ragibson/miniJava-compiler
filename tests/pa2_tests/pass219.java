// PA2 binop pass
class id {
    void p(){
        boolean x = true && false || x;
    }
}

