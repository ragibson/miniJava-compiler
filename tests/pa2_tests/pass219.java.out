********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "id" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     BOOLEAN BaseType
  .   .   .     "x" varname
  .   .   .   BinaryExpr
  .   .   .     "||" Operator
  .   .   .       BinaryExpr
  .   .   .         "&&" Operator
  .   .   .           LiteralExpr
  .   .   .             "true" BooleanLiteral
  .   .   .           LiteralExpr
  .   .   .             "false" BooleanLiteral
  .   .   .       RefExpr
  .   .   .         IdRef
  .   .   .           "x" Identifier
=============================================
